OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -394,22 +403,40 @@ - (void)track:(NSString *)eventKey
    attributes:(NSDictionary<NSString *,NSString *> *)attributes
     eventTags:(NSDictionary<NSString *,id> *)eventTags {
     
+    if ([Optimizely isEmptyString:eventKey]) {
+        [self handleErrorLogsForTrack:OPTLYLoggerMessagesTrackEventKeyEmpty ofLevel:OptimizelyLogLevelError];
+        return;
+    }
+    
+    if ([Optimizely isEmptyString:userId]) {
+        [self handleErrorLogsForTrack:OPTLYLoggerMessagesUserIdInvalid ofLevel:OptimizelyLogLevelError];
+        return;
+    }
+    
     OPTLYEvent *event = [self.config getEventForKey:eventKey];
     
     if (!event) {
-        [self handleErrorLogsForTrackEvent:eventKey userId:userId];
+        NSString *logMessage = [NSString stringWithFormat:OPTLYLoggerMessagesEventDispatcherEventNotTracked, eventKey, userId];
+        [self handleErrorLogsForTrack:logMessage ofLevel:OptimizelyLogLevelInfo];
         return;
     }
     
-    NSDictionary *conversionEventParams = [self.eventBuilder buildConversionTicket:self.config
-                                                                          bucketer:self.bucketer
-                                                                            userId:userId
-                                                                         eventName:eventKey
-                                                                         eventTags:eventTags
-                                                                        attributes:attributes];
+    NSArray *decisions = [self decisionsFor:event userId:userId attributes:attributes];
```

Comment:
Not sure what this is about, can you explain a bit more? It's a bit confusing that this happens in the top-level Optimizely class.

Reply:
This is about making decisions array containing experiments and their bucketed variations which are associated with an event.